client/src/ui/molecules/language-menu/index.tsx

```diff
@@ -38,6 +41,13 @@ export function LanguageMenu({
         // The default is the current locale itself. If that's what's chosen,
         // don't bother redirecting.
         if (localeURL !== locale) {
+          for (const translation of translations) {
+            if (translation.url === localeURL) {
+              document.cookie = `${PREFERRED_LOCALE_COOKIE_NAME}=${
+                translation.locale
+              };max-age=${60 * 60 * 24 * 365 * 3};path=/`;
```

Comment:
There are security implications about using non-HTTPS cookies but I can't imagine any possible way that you can abuse this. This code here is no different from using the Devtools and setting any cookie manually on developer.mozilla.org.

Reply:
can we add `;secure;samesite=lax`?

Reply:
> can we add `;secure;samesite=lax`?

Why? What is the point? 
How can you even get to the site with HTTP so that the browser might include this cookie? And even if you do, is there any difference?
And `SameSite=Lax` is the default, no?

Reply:
`SameSite=Lax`  is default indeed. Making it implicit is nice. I’ve encountered many bugs with implicit cookie defaults. 

Making it a secure cookie is definitely a best practice. I guess it would be submitted before https upgrade and if someone manages to serve something http on out behalf. 

I think they are mandatory for .mozilla.org sites (https://infosec.mozilla.org/guidelines/web_security.html)

Reply:
https cookies make sense for security, but we're not using cookies for something secure. We're using it like you use `localStorage` in the browser. If you set `secure` it tells the browser to not send it if the request is `http://` but even if you can trick someone to go to `http://developer.mozilla.org/docs/Web` you still haven't accomplished anything. 

I'm not questioning you, I'm questioning the idea/concept. 

If it causes a problem to use non-secure cookies it's easy to change. E.g.
```
if (!document.location.hostname !== 'localhost') { cookieValue += ';secure' }
```

Reply:
It seems inherently insecure because we're setting the cookie client-side, so to me it also doesn't seem worthwhile to add `secure`, but I guess I can't see how it would hurt either -- but maybe I'm missing something.